src/Agent.Worker/TestResults/Utils/FeatureFlagService.cs

```diff
@@ -44,9 +44,11 @@ public bool GetFeatureFlagState(string featureFlagName, Guid serviceInstanceId)
                 }
                 _executionContext.Debug(StringUtil.Format("{0} is off", featureFlagName));
             }
-            catch
+            catch(Exception ex)
             {
-                _executionContext.Debug(StringUtil.Format("Failed to get FF {0} Value.", featureFlagName));
+                var exceptionMessage = StringUtil.Format("Failed to get FF {0} Value. Error: {1}", featureFlagName, ex.ToString());
+                _executionContext.Warning(exceptionMessage);
```

Comment:
Shouldn't this be error log? why are we logging as warning

Reply:
I kept it as warning because of two reasons
  - This FF failed to fetch exception is not exactly a breaking scenario, from PTR/User point of view if FF failed to fetch FF will be treated as false and false workflow is executed. 
  - For an error scenario, we are generally expected to fail the task, however PTR is not expected to fail in this case.

So, I am logging this as warning along with the exception stack trace. let me know your thoughts on this.

Reply:
Is this warning message flowing to our Kusto for monitoring?

Reply:
Yes we are logging to producttrace